src/main/java/com/myhome/controllers/PaymentController.java

```diff
@@ -0,0 +1,184 @@
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * REST Controller which provides endpoints for managing payments
+ */
+@RestController
+@Slf4j
+public class PaymentController {
+  private final PaymentService paymentService;
+  private final CommunityService communityService;
+  private final SchedulePaymentApiMapper schedulePaymentApiMapper;
+
+  public PaymentController(
+          PaymentService paymentService,
+          CommunityService communityService,
+          SchedulePaymentApiMapper schedulePaymentApiMapper) {
+    this.paymentService = paymentService;
+    this.schedulePaymentApiMapper = schedulePaymentApiMapper;
+    this.communityService = communityService;
+  }
+
+  /*private Payment paymentDtoToPayment(PaymentDto payment) {
+    LocalDate date = LocalDate.parse(payment.getDueDate(), DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+    return new Payment(payment.getPaymentId(), payment.getCharge(), payment.getType(), payment.getDescription(), payment.isRecurring(), date, payment.getAdminId(), payment.getMemberId());
+  }*/
+
+  @Operation(description = "Schedule a new payment")
+  @PostMapping(
+          path = "/members/payments",
+          produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE},
+          consumes = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE}
+  )
+  public ResponseEntity<SchedulePaymentResponse> schedulePayment(@Valid @RequestBody
+                                                                   SchedulePaymentRequest request) {
+    log.trace("Received schedule payment request");
+    PaymentDto requestPaymentDto =
+        schedulePaymentApiMapper.schedulePaymentRequestToPaymentDto(request);
+    PaymentDto scheduledPayment = paymentService.schedulePayment(requestPaymentDto);
+
+    if (scheduledPayment == null) {
+      return  ResponseEntity.status(HttpStatus.NOT_FOUND).body(new SchedulePaymentResponse());
+    }
+
+    SchedulePaymentResponse scheduledPaymentResponse =
+        schedulePaymentApiMapper.paymentToSchedulePaymentResponse(scheduledPayment);
+
+    return ResponseEntity.status(HttpStatus.CREATED).body(scheduledPaymentResponse);
+  }
+
+  @Operation(description = "Get details about a payment with the given payment id")
+  @GetMapping(
+      path = "/members/payments/{paymentId}",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE}
+  )
+  public ResponseEntity<SchedulePaymentResponse> listPaymentDetails(
+     @PathVariable String paymentId) {
+    Optional<PaymentDto> paymentOptional = paymentService.getPaymentDetails(paymentId);
+    log.trace("Received request to get details about a payment with id[{}]", paymentId);
+
+    if (!paymentOptional.isPresent()) {
```

Comment:
Use `.map()` with `.orElseGet()`/`.orElse()` logic instead of `.ifPresent()`

Reply:
How does that work? I tried doing what you suggested but orElse tries to return a SchedulePaymentReposnse and not a ResponseEntity

Reply:
i did the [same thing](https://github.com/jmprathab/MyHome/blob/master/src/main/java/com/myhome/controllers/HouseMemberDocumentController.java#L91) may be this can help you

Reply:
Thanks @DDmit04 that worked :+1: